SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -67,11 +67,39 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
     }
 
     override fun setDepartureStation(departureStation: String) {
-        chosenDepartureStation = departureStation
+        chosenDepartureStation = matchStationNameToCode(departureStation)
     }
 
     override fun setArrivalStation(arrivalStation: String) {
-        chosenArrivalStation = arrivalStation
+        chosenArrivalStation = matchStationNameToCode(arrivalStation)
+    }
+
+    private fun buildStationDropdowns() {
+        launch {
+            val allStationDetails = getAllStationDetails()
+            buildStationNamesList(allStationDetails)
+            view!!.setDepartureDropdown(stationNames)
+            view!!.setArrivalDropdown(stationNames)
+        }
+    }
+
+    private fun buildStationNamesList(allStationDetails: StationDetails) {
+        for ((codeIndex, stationCode) in stationCodes.withIndex()) {
```

Comment:
You're no longer using the codeIndex variable here, so there's no need to use `withIndex` to iterate 🙂